### DISCUSSION TITLE
test command do not working

### DISCUSSION THREAD

**Discussion by khushbusandhu** [Cyfrin/foundry-full-course-cu#557] [Q&A]
i am getting an error while running ```forge test``` command
error is 
```[⠊] Compiling...
[⠒] Unable to resolve imports:
      "ds-test/test.sol" in "/home/khushbu/solidity-course/foundry-simple-storage-f23/foundry-funf-me-f23/lib/forge-std/src/StdAssertions.sol"
      "ds-test/test.sol" in "/home/khushbu/solidity-course/foundry-simple-storage-f23/foundry-funf-me-f23/lib/forge-std/src/Test.sol"
with remappings:
[⠢] Unable to resolve imports:/solidity-course/foundry-simple-storage-f23/foundry-funf-me-f23/lib/forge-std/src/
      "ds-test/test.sol" in "/home/khushbu/solidity-course/foundry-simple-storage-f23/foundry-funf-me-f23/lib/forge-std/src/StdAssertions.sol"
      "ds-test/test.sol" in "/home/khushbu/solidity-course/foundry-simple-storage-f23/foundry-funf-me-f23/lib/forge-std/src/Test.sol"
with remappings:
      forge-std/=/home/khushbu/solidity-course/foundry-simple-storage-f23/foundry-funf-me-f23/lib/forge-std/src/
[⠰] Compiling 21 files with 0.8.21
[⠔] Solc 0.8.21 finished in 142.55ms
Error: 
Compiler run failed:
Error (6275): Source "ds-test/test.sol" not found: File not found. Searched the following locations: "/home/khushbu/solidity-course/foundry-simple-storage-f23/foundry-funf-me-f23".
 --> lib/forge-std/src/StdAssertions.sol:4:1:
  |
4 | import {DSTest} from "ds-test/test.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "ds-test/test.sol" not found: File not found. Searched the following locations: "/home/khushbu/solidity-course/foundry-simple-storage-f23/foundry-funf-me-f23".
  --> lib/forge-std/src/Test.sol:27:1:
   |
27 | import {DSTest} from "ds-test/test.sol";
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^```

**Comment by cromewar** [comment]
Hello @khushbusandhu 
Can you please share also the code you are using for the test.

**Reply by khushbusandhu** [reply]
hey @cromewar! 
got the solution 
thanks for your gesture :)

**Reply by getzelus** [reply]
I had also this error, i changed the path to "lib/..." but test.sol didnt exist in my files so i found one online and put it but i had more erros... 
What is your solution ?

**Reply by ChanX21** [reply]
Could you explain the solution which worked for you @khushbusandhu as we are facing the same issue

**Reply by ShawnFromTexas** [reply]
Hello, did you ever find solution to this? running into same issue. thanks!

**Reply by ShawnFromTexas** [reply]
I solved my issue by installing forge-std again.

Try this...

```solidity
forge install foundry-rs/forge-std@v1.5.3 --no-commit
```


Cheers BUILDERS!

**Reply by DCVglobalnetwork** [reply]
doesnt working on Window11

**Comment by 0xKumarShivam** [comment]
use "is" 
```solidity
contract DemoTest is DSTest {
    // --- assertTrue ---  }
```
or

may be because you have empty lib/ds-test folder.

**Reply by cell0901** [reply]
exactly i was facing the same error, my folder was empty lib/ds-test was empty i have to manually search for it and paste it also i have to use this path "../lib/ds-test/src/test.sol", why it didn't come when i run forge init?

**Reply by getzelus** [reply]
It seems when you **clone** the git folder and run forge install, then it installs all.
Although if you download the zip folder, it wont work.

**Reply by ShawnFromTexas** [reply]
I believe my ds-test folder is empty as mine says folder not found. how do i solve this issue frens?

```solidity
Source "lib/ds-test/test.sol" not found: File not found. Searched the following locations: "".solidity(6275)
```

**Reply by ShawnFromTexas** [reply]
my entire code is here. many thanks!

https://github.com/Cyfrin/foundry-full-course-f23/discussions/1281

**Reply by ShawnFromTexas** [reply]
Solved issue by installing forge-std again 

```solidity
forge install foundry-rs/forge-std@v1.5.3 --no-commit
```

**Comment by getzelus** [comment]
I had also this error : test.sol was not found.
And I couldnt find the file in my folders (i have windows and used git bash to init foundry). 

- I found test.sol here : 
https://github.com/dapphub/ds-test/tree/e282159d5170298eb2455a6c05280ab5a73a4ef0

- Then in Test.sol, I changed the line for import to :  import {DSTest} from "../lib/ds-test/src/test.sol";

And now it works !

**Comment by ShawnFromTexas** [comment]
How did you fix this issue frens?
